src/Button/Button.tsx

```diff
@@ -276,293 +279,296 @@ export const Button = React.forwardRef<HTMLElement, Props>(
+                        color: getTextColor({
+                          color,
+                          feel,
+                          theme,
+                          mode: ":hover",
+                        }),
+                        ...(feel !== "flat" && {
+                          // The `box-shadow` property is copied directly from Zeplin
+                          boxShadow:
+                            theme === "light"
+                              ? "0 5px 10px 0 rgba(18, 21, 26, 0.08), inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)"
+                              : "0 0 0 1px rgba(18, 21, 26, 0.2), 0 5px 10px 0 rgba(18, 21, 26, 0.12), 0 1px 0 0 rgba(18, 21, 26, 0.05)",
+                        }),
+                      },
+                      // This is kind of hacky behavior
+                      "&[data-force-focus-state]": focusedStyles,
+                      "&:active, &[data-force-active-state], &[aria-expanded=true]": {
+                        ...(getTextColor({
+                          color,
+                          feel,
+                          theme,
+                          mode: ":hover",
+                        }) && {
+                          color: getTextColor({
+                            color,
+                            feel,
+                            theme,
+                            mode: ":active",
+                          }),
+                        }),
+
+                        backgroundColor:
+                          color === colors.white
+                            ? colors.white
+                            : feel === "raised"
+                            ? color
+                            : color === defaultColor
+                            ? theme === "dark"
+                              ? colors.grey.darker
+                              : colors.silver.base
+                            : getOffsetInPalette(2, "lighter", color),
+
+                        // The `box-shadow` properties are copied directly from Zeplin
+                        boxShadow:
+                          feel !== "flat"
+                            ? theme === "light"
+                              ? "inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05), inset 0 2px 2px 0 rgba(18, 21, 26, 0.12)"
+                              : "0 0 0 1px rgba(18, 21, 26, 0.2), 0 1px 4px 0 rgba(18, 21, 26, 0.08), 0 -1px 0 0 rgba(18, 21, 26, 0.16), inset 0 1px 2px 0 rgba(18, 21, 26, 0.42)"
+                            : "none",
+                        outline: "0",
+                      },
+                    },
+                  ])
+                )
+              ),
+              disabled,
+              onClick: (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
+                if (disabled) return event.preventDefault();
+
+                otherProps.onClick?.(event);
+                as.props.onClick?.(event);
```

Comment:
nbd: I wonder if we should throw when both are defined?

```tsx
<Button
  as={
    <button
      onClick={(e) => {
        e.preventDefault();
        e.stopPropagation();
      }}
    />
  }
  onClick={() => alert("expected behaviour seems unclear from reading usage site")}
/>;
```

or make it so that the click handler from the `as` component would override the one from the Button component (and warn that Button's will be ignored if both are provided?)

Reply:
I don't feel strongly about preventing `onClick` from going in either place. I'd honestly expect if it was in both that they would both work, but I don't feel super strongly about it.